src/forms/dispense-form.component.tsx

```diff
@@ -67,6 +77,30 @@ const DispenseForm: React.FC<DispenseFormProps> = ({
               medicationRequestBundle,
               config.dispenseBehavior.restrictTotalQuantityDispensed,
             );
+
+            if (config.enableStockDispense) {
```

Comment:
This is breaking up the logic around fulfiller status, can we break this up into a separate "then" block, or at least move it up to line 73, before the "const newFulfillerStatus..." line?

Also,   defer to you about the order operations... ie, would you want to move the dispense request *before* the "saveMedicationRequest" so that you *only* update the medication request if the stock update is successful?  Of course, then you run into the opposite risk: that the stock could be updated and then the dispense fails. Would be great to make this a little more transactional, but, admittedly, that is likely difficult.